### PR TITLE
Rename some legacy presto items to trino

### PR DESCRIPTION
**Pull request by dain**
## Release notes

( ) This is not user-visible or is docs only, and no release notes are required.
( ) Release notes are required. Please propose a release note for me.
(x) Release notes are required, with the following suggested text:

```markdown
# Hive
* Rename `presto_version` table property to `trino_version`. ({issue}`issuenumber`)
* Rename `presto_query_id` table property to `trino_query_id`. ({issue}`issuenumber`)
```

### CODE REVIEW

**Comment by wendigo** [comment]
Are these changes backward-compatible if some tables were created with the previous versions?

File: docs/src/main/sphinx/connector/hive.md [hunk 1]
```diff
@@ -909,9 +909,9 @@ SELECT * FROM example.web."page_views$properties";
 ```
 
 ```text
-       stats_generated_via_stats_task        | auto.purge |       presto_query_id       | presto_version | transactional
----------------------------------------------+------------+-----------------------------+----------------+---------------
- workaround for potential lack of HIVE-12730 | false      | 20230705_152456_00001_nfugi | 423            | false
+       stats_generated_via_stats_task        | auto.purge |       presto_query_id       | trino_version | transactional
+---------------------------------------------+------------+-----------------------------+---------------+---------------
+ workaround for potential lack of HIVE-12730 | false      | 20230705_152456_00001_nfugi | 423           | false
```
Comment (electrum) [inline docs/src/main/sphinx/connector/hive.md:914]: Update to latest Trino version

File: plugin/trino-hive/src/main/java/io/trino/plugin/hive/metastore/HivePrincipal.java [hunk 2]
```diff
@@ -110,9 +110,9 @@ public boolean equals(Object o)
         if (o == null || getClass() != o.getClass()) {
             return false;
         }
-        HivePrincipal prestoPrincipal = (HivePrincipal) o;
-        return type == prestoPrincipal.type &&
-                Objects.equals(name, prestoPrincipal.name);
+        HivePrincipal trinoPrincipal = (HivePrincipal) o;
```
Comment (electrum) [inline plugin/trino-hive/src/main/java/io/trino/plugin/hive/metastore/HivePrincipal.java:113]: This should probably be `hivePrincipal`.

**Review by electrum — APPROVED** [review]
We don't read these properties, so the change it safe. They are mainly intended for manual debugging or analysis.

File: docs/src/main/sphinx/release/release-0.153.md [hunk 3]
```diff
@@ -93,7 +93,7 @@ and {doc}`/admin/resource-groups` for more details.
   The filesystem location can be specified when creating a schema,
   which allows, for example, easily creating tables on S3.
 - Record Presto query ID for tables or partitions written by Presto
-  using the `presto_query_id` table or partition property.
+  using the `trino_query_id` table or partition property.
```
Comment (wendigo) [inline docs/src/main/sphinx/release/release-0.153.md:96]: that's pretty old release notes and this change isn't applicable to this version. revert

**Comment by mosabua** [comment]
Just to be certain.. this is NOT a breaking change, right @dain and @electrum ?

**Pull request merged**